core/src/test/java/hudson/util/io/ZipArchiverTest.java

```diff
@@ -102,6 +103,64 @@ public void backwardsSlashesOnWindows()  {
         assertEquals("foo/bar/baz/Test.txt", zipEntryName);
     }
 
+    @Test
+    public void huge64bitFile()  {
+        // create huge64bitFileTest.txt
+
+        File hugeFile = new File(tmpDir, "huge64bitFileTest.txt");
+        try {
+            RandomAccessFile largeFile = new RandomAccessFile(hugeFile, "rw");
+            largeFile.setLength(4* 1024 * 1024 * 1024 + 2);
+        } catch (IOException e) {
+            fail("unable to prepare source directory for zipping", e);
```

Comment:
IIUC we need 4GB of free disk space for this test to pass. Would it make sense to consider the test skipped, if that's possible, if we fail to create the file? Similar to OS dependent tests?

Reply:
Yes. When I set out to write this test, I planned to do that, but when I looked at the test that I stole from, I couldn't see a way to do that, so I left that part out.

Note: you don't precisely need 4gb of disk space free.

You need either:
* 4+gb of disk space free (space for the file, plus space for the archive), or
* a volume that supports sparse files, and a jenkins / java combination that are able to use that feature. I have no idea how often that will work out. (sparse file support exists for ntfs, zfs, apfs, ext4 and ufs, amongst others)